### ISSUE TITLE
Why white noise perturbations

### ISSUE THREAD

**Issue by moorepants** [csu-hmc/gait-control-direct-id-paper#47]
When asked how they would design a perturbation experiment, both Mont and Andy said they'd apply a single step or impulse perturbation in the stance phase with the belt at a specific point in the gait phase but apply in a random selection of the gait cycles. They expected that the mean gait cycle for perturbed gait cycles would have a different mean that those for unperturbed. Then you could subtract away the unperturbed mean to get the transient response. I think this was very similar to what Elliot Rouse did with his perturbation of the ankle. When I told them we applied random noise and that the mean gait cycle is the same for perturbed and unperturbed, they didn't think that was a good idea. Andy seemed to think it was blindly following the system identification recommendations. There is nothing to be done about this, but I just wanted to note it down.

**Comment by tvdbogert** [comment]
I don't agree.  Yes, a step or impulsive perturbation (if you do the same amplitude at the same time in all the perturbed gait cycles) would allow you to see the average response of all the perturbed cycles.  This would be nice and this is the classical neurophysiology approach to looking at motor control, reflexes etc.  I would certainly make the data presentation more appealing. But it would take forever to identify a MIMO time varying feedback network this way, and require special care to avoid bias due to insufficient variation in inputs and non-unique solutions. 

If the system is linear, the identification would not take forever with simple impulsive perturbations (since superposition holds).  The controller you are currently identifying could possibly be identified that way, and the fit would be great, but we know the real controller is more complex and we wanted to have the potential for our data to go beyond single joint linear controllers.

Non-unique solutions is a real risk.  If you push a person in a certain direction, you can't determine if a torque is responding to a change in head acceleration, ankle angular velocity, or any of the other sensory signals.  They are all very much correlated.  You can only separate this out if the data includes cases where head acceleration and ankle angle have a different relationship to one another.  So you need different types of perturbations, not just one.  If we are limited to horizontal pushes (belt accelerations), the best we can do is apply varying magnitude in both directions at random times.

We should not worry about the mean gait cycle.   Our perturbed gait cycles have the same mean as the unperturbed gait cycles, because we push in all possible directions at all possible times in the cycle.  I would be worried if the mean was not the same because it could bias the sys id results.

**Comment by moorepants** [comment]
> If the system is linear

Andy made the point that our system (the controller) is linear, just not time, or gait phase, invariant.

> Non-unique solutions is a real risk.

I agree with this reasoning. For a given MIMO system, applying identical perturbations in each gait cycle has less likelihood of exposing the response space.

I can't remember in detail what their arguments for these types of perturbations were any longer, they'd have to comment but I think it lied in ability to easily and explicitly expose the how the trajectories change with respect to the unperturbed trajectories.

> We should not worry about the mean gait cycle.

I'm not worried about it. It is what we expected given the applied perturbations.

I don't really agree with Mont and Andy here either but, like I said above, just wanted to note it down. I wish I'd taken better notes that detailed their specific concerns.
